Handle ending blank line.

weakish opened the pull request:
> The last record in the file may or may not have an ending line break.
> -- [RFC4180](https://tools.ietf.org/html/rfc4180)

Previous implementation converts a csv file with ending line break
to an HTML table with an additional empty row:

```
+-----+-------+
|Last | field |
+-----+-------+
|     |
+-----+
```

This commit correctly handles ending blank line,
i.e. converting to an HTML table without redundant empty row.

Permission to use, copy, modify, and/or distribute this commit
for any purpose with or without fee is hereby granted.

---

 CsvToTable.js | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

weakish committed:
Handle ending blank line.

> The last record in the file may or may not have an ending line break.
-- [RFC4180]

Previous implementation converts a csv file with ending line break
to an HTML table with an additional empty row:

    +-----+-------+
    |Last | field |
    +-----+-------+
    |     |
    +-----+

This commit correctly handles ending blank line,
i.e. converting to an HTML table without redundant empty row.

Permission to use, copy, modify, and/or distribute this commit
for any purpose with or without fee is hereby granted.

[RFC4180]: https://tools.ietf.org/html/rfc4180

yasharma commented:
To handle empty lines at the end as well as at any place, used array filter to filter through the array elements and removes empty elements.

```
// Removes empty elements from an array
    function clean(arr){
        arr.filter(function(value, index){
            var count = 0;
            if(value === "") {
                count++;
                arr.splice(index,count); 
            }
        });
    }
```

weakish committed:
Remove blank lines before convert.

According to [RFC4180], blank links are not allowed.
To be liberal in what we accept from others,
remove those blank lines before convert.

> file = [header CRLF] record *(CRLF record) [CRLF]

[RFC4180]: https://tools.ietf.org/html/rfc4180

weakish commented:
`.filter` is added in ECMAScript 5.1 (ECMA-262).
Using `filter` breaks Firefox <1.5 and IE <9.
Do I need to polyfill `filter` or not? (CsvToTable is supposed to work on recent versions of browsers or be compatible with old ones?)

yasharma commented:
Yes, it should also be compatible with old ones(IE8), we need polyfill for Array.filter

weakish committed:
polyfit .filter()

weakish commented:
Add a polyfit commit. Untested (no old browsers or cross browser testing framework on my machine.)

yasharma commented:
Merged Successfully.
